Src/LiquidProjections.NHibernate/NHibernateEventMapConfigurator.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LiquidProjections.NHibernate
+{
+    internal sealed class NHibernateEventMapConfigurator<TProjection, TKey>
+        where TProjection : class, IHaveIdentity<TKey>, new()
+    {
+        private readonly IEventMap<NHibernateProjectionContext> map;
+        private readonly IEnumerable<INHibernateChildProjector> children;
+
+        public NHibernateEventMapConfigurator(
+            IEventMapBuilder<TProjection, TKey, NHibernateProjectionContext> mapBuilder,
+            IEnumerable<INHibernateChildProjector> children = null)
+        {
+            if (mapBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(mapBuilder));
+            }
+
+            map = BuildMap(mapBuilder);
+            this.children = children?.ToList() ?? new List<INHibernateChildProjector>();
+        }
+
+        private IEventMap<NHibernateProjectionContext> BuildMap(
+            IEventMapBuilder<TProjection, TKey, NHibernateProjectionContext> mapBuilder)
+        {
+            mapBuilder.HandleCustomActionsAs((_, projector) => projector());
+
+            mapBuilder.HandleProjectionModificationsAs(async (key, context, projector, options) =>
+            {
+                switch (options.MissingProjectionBehavior)

```

Comment:
Should we maybe avoid using behaviors then? It feels weird that we first convert different fluent methods into the same delegate with different behaviors but then convert back to different methods.